src/averageAges.js

```diff
@@ -1,61 +1,64 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const isMale = person => person.sex === 'm';
+const turnToCentury = deathDate => Math.ceil(deathDate / 100);
+const calculateAge = person => person.died - person.born;
+const calculateTotalAgeOfGroup = (sum, person) => sum + person;
+const takeAverage = value => Math.round(value * 100) / 100;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? isMale(person) && turnToCentury(person.died) === century
+    : isMale(person)
+  );
+
+  const menAge = men.map(calculateAge);
+
+  const totalAge = menAge.reduce(calculateTotalAgeOfGroup);
+
+  const averageAge = takeAverage(totalAge / menAge.length);
+
+  return averageAge;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
+const isFemale = person => person.sex === 'f';
+const isMother = (mother, child) => mother.name === child.mother;
```

Comment:
Could I suggest grouping these 2 global variables with the other global variables at the top.